pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Options.groovy

```diff
@@ -72,31 +83,52 @@ public class Options implements Serializable {
     private static final LoadingCache<Object,Map<String,String>> optionTypeCache =
         Utils.generateTypeCache(DeclarativeOptionDescriptor.class, false, [])
 
+    public static Map<String,String> getEligibleWrapperStepClasses() {
+        Map<String,String> stepNames = [:]
+
+        // TODO: Figure out if we want to support metasteps? Don't think so.
+        ExtensionList.lookup(StepDescriptor.class).each { s ->
+            if (s.takesImplicitBlockArgument()
+                && !(s.getFunctionName() in ModelASTMethodCall.blockedSteps.keySet())
+                && !(Launcher.class in s.getRequiredContext())
+                && !(FilePath.class in s.getRequiredContext())) {
```

Comment:
I'm not sure about the cost of calling `s.getRequiredContext()` but if it does some kind of lookup you should consider cashing it in a local variable first.